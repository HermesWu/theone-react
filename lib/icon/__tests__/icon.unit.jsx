import * as renderer from 'react-test-renderer';
import * as React from 'react';
import Icon from '../icon';
import {mount} from 'enzyme'

describe('Icon', () => {
  it('render successfully', () => {
    const json = renderer.create(<Icon name='wechat'/>).toJSON();
    expect(json).toMatchSnapshot();
  });
  it('onClick', () => {
    const fn = jest.fn()
    const fn2 = jest.fn()
    const component = mount(<Icon name="wechat" onClick={fn}/>)
    component.find('svg').simulate('click')
    expect(fn).toBeCalled()
  })
});