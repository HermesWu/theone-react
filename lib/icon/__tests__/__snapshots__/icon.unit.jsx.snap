// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon render successfully 1`] = `
<svg
  className="theone-icon"
>
  <use
    xlinkHref="#wechat"
  />
</svg>
`;
